src/components/atoms/DownloadPDFLink/DownloadPDFLink.ts

```diff
@@ -0,0 +1,108 @@
+import { html, LitElement } from '@polymer/lit-element';
+import CSS from './_components.download-pdf-link-css';
+import getText from '../../../i18n/getText';
+import { TemplateResult } from 'lit-html';
+import { CONTENT_TYPES } from '../../../constants/contentTypes';
+import domain from '../../../domain';
+
+export interface IDownloadPDFLinkApi {
+  display?: string;
+  isVisible?: boolean;
+  contentType?: CONTENT_TYPES;
+  contentEncoding?: string;
+  content?: string;
+  recipientName?: string;
+  certificateTitle?: string;
+  issueDate?: string;
+  issuerName?: string;
+  issuerLogo?: string;
+  recordLink?: string;
+  issuerPublicKey?: string;
+}
+
+class DownloadPDFLink extends LitElement {
+  constructor () {
+    super();
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  static get properties (): any {
+    return {
+      display: String,
+      isVisible: Boolean,
+      contentType: String,
+      contentEncoding: String,
+      content: String,
+      recipientName: String,
+      certificateTitle: String,
+      issueDate: String,
+      issuerName: String,
+      issuerLogo: String,
+      recordLink: String,
+      issuerPublicKey: String
+    };
+  }
+
+  _render (props: IDownloadPDFLinkApi): TemplateResult {
+    if (!props.isVisible) {
+      return null;
+    }
+
+    const isPlainText = props.display === 'plaintext';
+    const label: string = getText('text.downloadPDFLink');
+    const labelIsLoading: string = getText('text.downloadPDFLinkIsLoading');
+
+    const classes = [
+      'buv-o-button-link',
+      'buv-c-download-pdf-link',
+      isPlainText ? '' : 'buv-c-download-pdf-link__icon'
+    ].join(' ');
+
+    return html`
+      ${CSS}
+      <button onclick='${async () => { await this.handleClick(props); }}'
+              class$='${classes}'
+              title$='${label}'>
+        <label class$='buv-c-download-pdf-link__label ${isPlainText ? 'buv-o-button-link__label' : 'buv-u-visually-hidden'}'>
+          <span class='buv-c-download-pdf-link__label--static'>${label}</span>
+          <span class='buv-c-download-pdf-link__label--loading'>${labelIsLoading}</span>
+        </label>
+      </button>`;
+  }
+
+  toggleButtonLoading (isLoading: boolean) {
+    const buttonElement: HTMLElement = this.shadowRoot.querySelector('.buv-c-download-pdf-link');
+    buttonElement.classList.toggle('buv-c-download-pdf-link--loading', isLoading);
+  }
+
+  async handleClick (props: IDownloadPDFLinkApi): Promise<void> {
+    this.toggleButtonLoading(true);
+    await domain.certificates.downloadPDF(props);
```

Comment:
I thought about this, but I need to toggle the loading state before and on completion of the generation of the PDF. There's probably a way to do it with an action but I honestly just wanted to get it done in this iteration.

Reply:
yeah I understand but this is breaking separation of concerns and creating an unexpected behavior with the rest of the code. Of all the comments this might be the most important one in terms of code quality and scalability. 
I don't think you need any reference to the state, nor a reducer, just an action that calls this.